### PR TITLE
Slow geoserver connections

### PR DESCRIPTION
**Pull request by akater320**
PiazzaEnvironment now creates its own instance of RestTemplate and can set a non-default timeout.

### CODE REVIEW

File: src/test/java/access/test/PiazzaEnvironmentTests.java [hunk 1]
```diff
@@ -15,63 +15,277 @@
...
+                Mockito.eq(String.class)))
+                .thenThrow(new HttpClientErrorException(HttpStatus.INTERNAL_SERVER_ERROR))
+                .thenThrow(new RuntimeException());
+
+        this.piazzaEnvironment.initializeEnvironment();
+        this.piazzaEnvironment.initializeEnvironment();
+        this.piazzaEnvironment.initializeEnvironment();
+    }
+
+//    @Test
```
Comment (go0ty) [inline src/test/java/access/test/PiazzaEnvironmentTests.java]: Remove commented , unused code

File: src/test/java/access/test/PiazzaEnvironmentTests.java [hunk 2]
```diff
@@ -15,63 +15,277 @@
...
+        ReflectionTestUtils.setField(this.piazzaEnvironment, "postgresDatabase", "junit_database_name");
+        ReflectionTestUtils.setField(this.piazzaEnvironment, "postgresHost", "junit_host");
+
+        Mockito.when(this.accessUtilities.getGeoServerBaseUrl()).thenReturn("http://localhost:8080/geoserver");
+    }
+
+    @Test
+    public void testResourcesExist() {
+        Mockito.when(this.restTemplate.exchange(
+                Mockito.endsWith("/workspaces/piazza.json"),
```
Comment (go0ty) [inline src/test/java/access/test/PiazzaEnvironmentTests.java:83]: nice use of "endsWith"!

File: src/main/java/access/deploy/geoserver/PiazzaEnvironment.java [hunk 3]
```diff
@@ -201,7 +214,7 @@ private void createWorkspace() {
 	/**
 	 * Creates the Piazza Postgres vector data store
 	 */
-	private void createPostgresStore() {
+	private void createPostgresStore(RestTemplate restTemplate) {
```
Comment (go0ty) [inline src/main/java/access/deploy/geoserver/PiazzaEnvironment.java]: What's the reasoning for passing the class-member variable via a method parameter? Do you want to do this vs. just using the member variable reference?

**Review by go0ty — CHANGES_REQUESTED** [review]

File: src/main/java/access/deploy/geoserver/PiazzaEnvironment.java (hunk 3 above)
Comment (akater320) [reply src/main/java/access/deploy/geoserver/PiazzaEnvironment.java]: Good catch. That was left over from when the RestTemplate was stack-scoped.
Should be fixed now.

File: src/test/java/access/test/PiazzaEnvironmentTests.java (hunk 1 above)
Comment (akater320) [reply src/test/java/access/test/PiazzaEnvironmentTests.java]: Good call. Done.

**Review by go0ty — APPROVED** [review]

**Pull request merged**